[Ang Ping Young] iP

kynapy opened the pull request:
## Greg

> Road work ahead? Uh, yeah, I sure hope it does! - Drew Gooden [[source](https://www.youtube.com/watch?v=6AYv6rV3NXE)]

Greg is your friendly companion and helps you remember your important tasks! It's:
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is:
1. download it from [here](https://github.com/kynapy/ip/releases/download/A-Jar/duke-v0.1.jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😄 

And it is **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

<blockquote>:)</blockquote>

##

Here is the ```main``` method of the Duke program:
```java
public class Duke {
	public static void main(String[] args) {
		new Duke("./data/duke.txt").run();
	}
}
```

kynapy committed:
Level-1

kynapy committed:
Level-2

kynapy committed:
Level-3

kynapy committed:
Add ToDo, Event and Deadline functionality

kynapy committed:
Automate Testing

kynapy committed:
edit expected output

kynapy committed:
Add functionality to delete tasks

sprintaway commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,131 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    public static void main(String[] args) throws Exception{
```
Perhaps a more specific exception should be thrown instead of a general exception?

sprintaway commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,131 @@
+                tasks.get(taskNumber).setStatus(1);
+                System.out.println("\n" + spacing);
+                System.out.println("Nice! I have marked this task as done:");
+                System.out.println("[" + tasks.get(taskNumber).getStatusIcon() +"] " + tasks.get(taskNumber).getTask());
```
Could toString be used here instead (also for unmark)?

sprintaway commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,131 @@
+
+            // Deadline
+            else if (word.startsWith("deadline")) {
+                String[] deadlineTask = (word.replace("deadline ", "")).split("/by ");
```
Perhaps the extra brackets (before word.replace) could be removed?

sprintaway commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,131 @@
+                    System.out.println("List empty. Add tasks into your list!\n");
+                    System.out.println(spacing + "\n");
+                }
+                else { 
```
Perhaps it should be } else { instead of current formatting?

sprintaway reviewed (commented):
LGTM apart from some small nits. Overall tasks classes are separated well and code is easy to read.

riccqi commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+    
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by + ")";
```
We could use the `String.format()` function in java to make this expression abit more readable

riccqi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,131 @@
+            String word = br.readLine();
+            // 1. Say goodbye
+            if (word.equals("bye")) { 
+                System.out.println("\n" + spacing);
```
We can try abstracting these repeated code to a separate function/class

riccqi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,131 @@
+            }
+
+            // Invalid input
+            else {
```
formatting nit: this else statement should be together with line 121!

riccqi reviewed (commented):
LGTM! Easy to read, and your formatting errors can be easily fixed by enabling auto format on save in intellij. [This](https://www.jetbrains.com/help/idea/reformat-and-rearrange-code.html) may be useful for that

kynapy committed:
Add functionality to load and save tasklist

kynapy committed:
Replace use of String literal with LocalDate and LocalTime classes for Deadline and Event timing

kynapy committed:
Merge branch 'branch-Level-8' into master

kynapy committed:
Fix bugs after merge of branch-Level-8 into master

kynapy committed:
Add additional classes for OOP, not finished

kynapy committed:
Change code to fit OOP

kynapy committed:
Organise into packages

kynapy committed:
Add Gradle

kynapy committed:
Add JUnit test cases for Todo and TaskList

kynapy committed:
Fix bugs in TaskList and TaskListTest

kynapy committed:
Add JavaDocs

kynapy committed:
Fix code to match coding standards

kynapy committed:
Add functionality to look for tasks

kynapy committed:
Fix code to match coding standards

kynapy committed:
Add functionality to look for tasks with keywords

kynapy committed:
Add DukeException class

kynapy committed:
Update package for DukeException

kynapy committed:
Add files and configuration for checkstyle

kynapy committed:
Add GUI for Duke

kynapy committed:
Delete .vscode directory

kynapy committed:
Fix build.gradle main method

kynapy committed:
Merge branch 'master' of https://github.com/kynapy/ip

kynapy committed:
Add assertions

kynapy committed:
Change code to improve code quality.
Multiple unary operators were found without a space after them. Trailing whitespace between code and comments are removed.
Spelling errors within JavaDoc were found. Certain class attributes could have their access modifiers
set to private for security. Code could be shortened to become more concise.
The above listed problems have now been updated. Code can be further shortened for methods that
are only single lined.

kynapy committed:
Merge pull request #2 from kynapy/branch-A-Assertions

kynapy committed:
Merge pull request #3 from kynapy/branch-A-CodeQuality

kynapy committed:
Add functionality to track expenses.

Expense tracking is now available. Expenses are stored in the
same data file as the Tasks to keep data in a single file. Expense and
ExpenseList classes are added for the Expense tracking functionality.
Storage class is modified to load and store Expenses, Parser is modified
to be able to parse commands related to Expenses. Bye command now
properly terminates Greg.

kynapy committed:
Add screenshot of Greg's GUI

kynapy committed:
Fix bug in termination of Greg

kynapy committed:
Add more exceptions and help command

Introduced a new InvalidInputException class to handle invalid
inputs from user. More errors are now handled with exceptions.
More problematic user inputs are now being handled, such as inputting
Strings or blanks for positions, or if the user does not input certain
fields required for the command. A new help command is added to
provide the user with a list of available commands.

kynapy committed:
Merge branch 'master' of https://github.com/kynapy/ip

kynapy committed:
Add greetings and fix coding standard

kynapy committed:
Modify README.md file to add user guide

kynapy committed:
Update README.md

kynapy committed:
Update README.md

kynapy committed:
Update README.md
